src/app/services/params.service.ts

```diff
@@ -0,0 +1,98 @@
+import { inject, Injectable } from '@angular/core';
+import { DateFormat, DateRange } from '../models';
+import { QueryParams } from '../models/query-params.model';
+import { DateService } from './date';
+
+/**
+ * Gets and sets params without changing navigation history.
+ * Because this, it doesn't work with the `Router` Service.
+ */
+@Injectable({
+	providedIn: 'root',
+})
+export class ParamsService {
```

Comment:
Initially I went with avoiding `Router` all together since I didn't want query params to not trigger navigations, but I believe this can be achieved using `Router` with `skipLocationChange` or whatever navigation option that doesn't update navigation.